File: add/codesnippet/CPP/t-system.net.icredential_1.cpp
public ref class SelectedHostsCredentialPolicy: public ICredentialPolicy
{
public:
   SelectedHostsCredentialPolicy(){}

   virtual bool ShouldSendCredential( Uri^ challengeUri, WebRequest^ request, NetworkCredential^ /*credential*/, IAuthenticationModule^ /*authModule*/ )
   {
      Console::WriteLine( L"Checking custom credential policy." );
      if ( request->RequestUri->Host->Equals( L"www.contoso.com" ) || challengeUri->IsLoopback == true )
            return true;

      return false;
   }
};

File: add/codesnippet/CSharp/t-system.net.icredential_1.cs
    public class SelectedHostsCredentialPolicy: ICredentialPolicy
    {
        public SelectedHostsCredentialPolicy()
        {
        }
        
        public virtual bool ShouldSendCredential(Uri challengeUri, 
            WebRequest request, 
            NetworkCredential credential, 
            IAuthenticationModule authModule)
        {
            Console.WriteLine("Checking custom credential policy.");
            if (request.RequestUri.Host == "www.contoso.com" ||
                challengeUri.IsLoopback == true)
                return true;

            return false;
        }
    }